File: src/jvmMain/kotlin/com/dropbear/input/InputStateNative.java
package com.dropbear.input;

import com.dropbear.EucalyptusCoreLoader;
import com.dropbear.math.Vector2d;

public class InputStateNative {
    static {
        new EucalyptusCoreLoader().ensureLoaded();
    }

    public static native void printInputState(long inputStateHandle);
    public static native boolean isKeyPressed(long inputStateHandle, int keyCode);
    public static native Vector2d getMousePosition(long inputStateHandle);
    public static native boolean isMouseButtonPressed(long inputStateHandle, MouseButton mouseButton);
    public static native Vector2d getMouseDelta(long inputStateHandle);
    public static native boolean isCursorLocked(long inputStateHandle);
    public static native void setCursorLocked(long commandBufferPtr, long inputStateHandle, boolean locked);
    public static native Vector2d getLastMousePos(long inputStateHandle);
    public static native boolean isCursorHidden(long inputStateHandle);
    public static native void setCursorHidden(long commandBufferPtr, long inputStateHandle, boolean hidden);
    public static native long[] getConnectedGamepads(long inputStateHandle);
}

File: src/nativeMain/kotlin/com/dropbear/input/InputState.native.kt
package com.dropbear.input

import com.dropbear.math.Vector2d

actual class InputState actual constructor() {
    internal actual fun printInputState() {
    }

    internal actual fun isKeyPressed(key: KeyCode): Boolean {
        TODO("Not yet implemented")
    }

    internal actual fun getMousePosition(): Vector2d {
        TODO("Not yet implemented")
    }

    internal actual fun isMouseButtonPressed(button: MouseButton): Boolean {
        TODO("Not yet implemented")
    }

    internal actual fun getMouseDelta(): Vector2d {
        TODO("Not yet implemented")
    }

    internal actual fun isCursorLocked(): Boolean {
        TODO("Not yet implemented")
    }

    internal actual fun setCursorLocked(locked: Boolean) {
    }

    internal actual fun getLastMousePos(): Vector2d {
        TODO("Not yet implemented")
    }

    internal actual fun isCursorHidden(): Boolean {
        TODO("Not yet implemented")
    }

    internal actual fun setCursorHidden(hidden: Boolean) {
    }

    internal actual fun getConnectedGamepads(): List<Gamepad> {
        TODO("Not yet implemented")
    }
}